src/oas/transformers/schema/__tests__/schema.spec.ts

```diff
@@ -0,0 +1,174 @@
+import { translateSchemaObject } from '..';
+import { OASSchemaObject } from '../types';
+
+const translate = (schemaObject: OASSchemaObject) => translateSchemaObject(schemaObject, { pruneNotSupported: [] });
+
+describe('translateSchemaObject', () => {
+  it('should translate id', () => {
+    expect(
+      translate({
+        type: 'object',
+        properties: {
+          test: {
+            id: 'test',
+          },
+        },
+      }),
+    ).toStrictEqual({
+      $schema: 'http://json-schema.org/draft-07/schema#',
+      type: 'object',
+      properties: {
+        test: {
+          $id: 'test',
+        },
+      },
+    });
+  });
+
```

Comment:
Just in case, I would consider adding a test to make sure passed schema is really not mutated.

```suggestion
it('should not mutate schema', () => {
    let before: OASSchemaObject = {
      type: 'object',
      properties: {
        test: {
          id: 'test',
        },
      },
    };

    translate(before);

    expect(before).toStrictEqual({
      type: 'object',
      properties: {
        test: {
          id: 'test',
        },
      },
    });
  });
```